index.md

```diff
@@ -1269,133 +1299,199 @@ def is_int(obj):
     return isinstance(obj, int)
 ```
 
+This error is similar to [C0121](#C0121).
 
-### Dangerous default value (W0102) {#W0102}
 
-This error occurs when a mutable value such as a list or dictionary is given a
-default value in the function or method definition. It is dangerous to give
-mutable objects a default value only when the function/method modifies the
-argument. If you modify a default argument, those changes will remain for the
-next time the function is called.
+### Dangerous default value (W0102) {#W0102}
 
-Hence your "empty" list or dictionary will start to contain values on calls
-other than the first call.
+This warning occurs when a mutable object, such as a list or dictionary, is provided as a default argument inside a function or a method definition. Default arguments are instantiated only once, at the time when the function or the method is defined (i.e. when the interpreter parses the `def ...` block). If the default argument is modified, it will remain modified in all subsequent calls of the function or method. This leads to a common "gotcha" in Python, where an "empty" list or dictionary, specified as the default argument, starts containing values on calls other than the first call.
 
 ~~~~ {include="W0102_dangerous_default_value"}
 ~~~~
 
-Though the output to this is be expected to be:
+Many new users of Python would expect the output of the code above to be:
 
 ```
 [0, 1, 2, 3, 4]
 [0, 1, 2, 3, 4]
 ```
 
-The actual output is:
+However, the actual output is:
 
 ```
 [0, 1, 2, 3, 4]
 [0, 1, 2, 3, 4, 0, 1, 2, 3, 4]
 ```
 
-If you want to avoid this situation then, you should use `None` as a default
-value, and then check for this default value inside the function body.
+If you want to prevent this surprising behavior, you should use `None` as the default argument, and then check for `None` inside the function body. For example, the following code prints the expected output:
+
+```python
+def make_list(n, lst=None):
+    if lst is None:
+        lst = []
+    for i in range(n):
+        lst.append(i)
+    print(lst)
```

Comment:
You should use `return` here. (Otherwise this is inconsistent with the next two lines.)